src/main/java/io/cdap/plugin/gcp/speech/SpeechToTextTransform.java

```diff
@@ -313,50 +324,80 @@ public String getLanguage() {
       return language == null ? "en-US" : language;
     }
 
+    /**
+     * Returns sample rate as an integer.
+     *
+     * @throws IllegalArgumentException if sample rate is an invalid number
+     */
     @Nullable
     public Integer getSampleRate() {
       if (containsMacro("sampleRate")) {
         return null;
       }
+      try {
+        Integer.parseInt(sampleRate);
+      } catch (NumberFormatException e) {
+        throw new IllegalArgumentException("Sample rate should be a valid number");
+      }
       return Integer.parseInt(sampleRate);
     }
 
-    private void validate(@Nullable Schema inputSchema) {
+    private void validate(@Nullable Schema inputSchema, FailureCollector collector) {
       if (inputSchema != null) {
         // If a inputSchema and audioField is available then verify that the schema does contain the given audioField
         // and that the type is byte array. This will allow to fail fast i.e. during deployment time.
         String audioFieldName = getAudioField();
-        if (audioFieldName != null && inputSchema.getField(audioFieldName) == null) {
-          throw new IllegalArgumentException(String.format("Field %s does not exists in the input schema.",
-                                                           audioFieldName));
-        }
         Schema.Field field = inputSchema.getField(audioFieldName);
-        Schema fieldSchema = field.getSchema();
-        Schema.Type fieldType =
-          fieldSchema.isNullable() ? fieldSchema.getNonNullable().getType() : fieldSchema.getType();
-        if (fieldType != Schema.Type.BYTES) {
-          throw new IllegalArgumentException(String.format("Field '%s' must be of type 'bytes' but is of type " +
-                                                             "'%s'.", field.getName(), fieldType));
+        if (audioFieldName != null && field == null) {
+          collector.addFailure(
+            String.format("Field '%s' does not exist in the input schema.", audioFieldName),
+            String.format("Field '%s' must exist in the input schema.", audioFieldName))
```

Comment:
This error message and corrective action is basically the same. In those situations, I think it's better to just have an error message and no corrective action, otherwise it's basically repeating the same thing.